app/src/main/kotlin/com/simplemobiletools/notes/pro/fragments/ChecklistFragment.kt

```diff
@@ -24,93 +22,126 @@ import com.simplemobiletools.notes.pro.models.Note
 import kotlinx.android.synthetic.main.fragment_checklist.view.*
 
 class ChecklistFragment : NoteFragment(), ChecklistItemsListener {
-    private var noteId = 0L
+
+    private var noteId      = 0L
+    private var items       = ArrayList<ChecklistItem>()
     private var note: Note? = null
-    private var items = ArrayList<ChecklistItem>()
 
     lateinit var view: ViewGroup
 
+    val checklistItems get(): String = Gson().toJson(items)
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        view = inflater.inflate(R.layout.fragment_checklist, container, false) as ViewGroup
+        view   = inflater.inflate(R.layout.fragment_checklist, container, false) as ViewGroup
         noteId = arguments!!.getLong(NOTE_ID, 0L)
         return view
     }
 
     override fun onResume() {
         super.onResume()
 
-        NotesHelper(activity!!).getNoteWithId(noteId) {
-            if (it != null && activity?.isDestroyed == false) {
-                note = it
+        loadNoteById(noteId)
+    }
+
+    override fun setMenuVisibility(menuVisible: Boolean) {
+        super.setMenuVisibility(menuVisible)
+
+        if (menuVisible) activity?.hideKeyboard()
```

Comment:
do not use such one liners please, use brackets at these cases for better readability